src/main/java/com/salesforce/mirus/MirusSourceTask.java

```diff
@@ -215,6 +228,22 @@ public void commit() {
     successfulCommitTime = time.milliseconds();
   }
 
+  /**
+   * Callback that is called when a record is committed.
+   */
+  @Override
+  public void commitRecord(SourceRecord record, RecordMetadata metadata) {
+      //
+      // record.kafkaPartition() is null when 1:1 partition mapping is not enabled.
+      //
+      try {
+        long latency = System.currentTimeMillis() - metadata.timestamp();
+        mirrorJmxReporter.recordMirrorLatency(record.topic(), latency);
```

Comment:
Any risk of a performance hit here? We will have multiple threads calling synchronized method all within a tight loop? Do we need to test for a performance cost?

Reply:
I don't think there is a performance implication. However, `commitRecord()` is called in the [send record method](https://github.com/apache/kafka/blob/2.6.1/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java#L343-L380), so it could in theory cause performance implications.

Reply:
Let's be aware of this and actively look for perf issues as we deploy.